src/sonic-config-engine/sonic_device_util.py

```diff
@@ -27,7 +30,45 @@ def get_machine_info():
             if len(tokens) < 2:
                 continue
             machine_vars[tokens[0]] = tokens[1].strip()
-    return machine_vars
+    return machine_vars 
+
+def get_npu_id_from_name(npu_name):
+    if npu_name.startswith(NPU_NAME_PREFIX):
+        return npu_name[len(NPU_NAME_PREFIX):]
+    else:
+        return None
+
+def get_num_npus():
+   platform = get_platform_info(get_machine_info())
+   asic_conf_file_path = os.path.join(SONIC_DEVICE_PATH, platform, 'asic.conf')
+   if not os.path.isfile(asic_conf_file_path):
+        return 1
+   with open(asic_conf_file_path) as asic_conf_file:
+	for line in asic_conf_file:
+	    tokens = line.split('=')
+            if len(tokens) < 2:
+               continue      
+	    if tokens[0].lower() == 'num_asic':
```

Comment:
mixed tab and spaces, suggest to use retab to correct all of them.

Reply:
Fixed in latest commit